case-lib/opt.sh

```diff
@@ -115,27 +116,13 @@ func_opt_parse_option()
         fi
     done
     
-    [ ${OPT_VALUE_lst['h']} -eq 1 ] && _func_opt_dump_help
-    # verify & check all option values
-    [[ $(declare -p |grep "[[:space:]]OPT_VALUE_lst=\"\"") ]] && echo "Missing parameter/script default value" && _func_opt_dump_help
-    # record the full parameter to the cmd
-    if [ ! -f $LOG_ROOT/cmd.txt ]; then
-        local cmd="$0" opt
-        for opt in ${!OPT_OPT_lst[@]}
-        do
-            [[ $opt == "h" ]] && continue
-            if [ ${OPT_PARM_lst[$opt]} -eq 1 ]; then
-                cmd="$cmd -$opt ${OPT_VALUE_lst[$opt]}"
-            else
-                if [ $(echo '!'${OPT_VALUE_lst[$opt]}|bc) -eq 1 ]; then
-                    cmd="$cmd -$opt"
-                fi
-            fi
-        done
-        echo $cmd >> $LOG_ROOT/cmd.txt
-        local git_path=$(dirname $0)
-        echo "Branch: "$(git -C $git_path branch |grep '^*') >> $LOG_ROOT/version.txt
-        echo "Commit: "$(git -C $git_path log --oneline -1) >> $LOG_ROOT/version.txt
+    [ "${OPT_VALUE_lst['h']}" -eq 1 ] && _func_opt_dump_help
```

Comment:
To be compatible with a future `set -e` and to look more like an assert:
```shell
[ "${OPT_VALUE_lst['h']}" -ne 1 ] || _func_opt_dump_help
```

Reply:
do you know which `exit` is load by `set -e`  
we hijack the `exit` function: https://github.com/thesofproject/sof-test/blob/master/case-lib/hijack.sh#L6  
so we can keep test-case more clean and easy to understand  
when case want to finished, it just need to `exit`

Reply:
> we hijack the exit function

Ouch... :-(

I just did a quick test (`( set -e ; false; exit 2 ; )`) and `set -e` apparently sticks to the native exit(), it ignores the "hijack".

To clean up on exit use `trap sof_test_cleanup EXIT` https://mywiki.wooledge.org/SignalTrap#Examples

This is compatible with `set -e`, demonstration:
```
( exit_sof_test() { printf "MYEXIT"; }; trap exit_sof_test EXIT;    set -e; false )
```
Never redefine built-ins, this is very confusing.

Reply:
when `trap` the `exit` signal will effect `exit` in `tools` ?
I know the `trap` will hijack signal but here also will effect script  
we are not put each `bash function` into `case-lib`  
we also think have some standalone `bash function`  
so we push them into `tools`